tradeit-android-sdk/src/main/java/it/trade/android/sdk/model/TradeItLinkedBrokerParcelable.java

```diff
@@ -37,13 +57,91 @@ protected void cache() {
         linkedBrokerCache.cache(this);
     }
 
-//    public void refreshBalancesForAllAccounts() {
-//
-//    }
+    public void refreshAccountBalances(final TradeItCallBackCompletion callback) {
+        RxJavaPlugins.setErrorHandler(new Consumer<Throwable>() {
+            @Override
+            public void accept(@NonNull Throwable e) throws Exception {
+                if (e instanceof UndeliverableException) {
+                    e = e.getCause();
+                }
+                if ((e instanceof IOException) || (e instanceof SocketException)) {
+                    // fine, irrelevant network problem or API that throws on cancellation
+                    return;
+                }
+                if (e instanceof InterruptedException) {
+                    // fine, some blocking code was interrupted by a dispose call
+                    return;
+                }
+                if ((e instanceof NullPointerException) || (e instanceof IllegalArgumentException)) {
+                    // that's likely a bug in the application
+                    Thread.currentThread().getUncaughtExceptionHandler()
+                            .uncaughtException(Thread.currentThread(), e);
+                    return;
+                }
+                if (e instanceof IllegalStateException) {
+                    // that's a bug in RxJava or in a custom operator
+                    Thread.currentThread().getUncaughtExceptionHandler()
+                            .uncaughtException(Thread.currentThread(), e);
+                    return;
+                }
+                Log.w(TAG, "Undeliverable exception received, not sure what to do", e);
+            }
+        });
+        Observable.fromArray(this.accounts)
+                .observeOn(AndroidSchedulers.mainThread(), true)
+                .subscribeOn(Schedulers.io())
+                .flatMapIterable(new Function<List<TradeItLinkedBrokerAccountParcelable>, Iterable<TradeItLinkedBrokerAccountParcelable>>() {
```

Comment:
I'm confused why we need `flatMap` here. It seems to not do anything; it just takes the list of accounts and returns it...same with the `authenticateAll`...